fix(app): run pages deploy workflow contract under the declared Vitest lane

## Problem

Current `develop@0b863b695bddb36708f39bb57a2d0b520f6d19da` fails the authoritative client lane in `packages/app/test/pages-deploy-workflow.test.ts` before collecting any tests:

```text
FAIL test/pages-deploy-workflow.test.ts [ test/pages-deploy-workflow.test.ts ]
Error: Cannot bundle built-in module "bun:test" imported from "test/pages-deploy-workflow.test.ts".
Test Files 1 failed | 88 passed (89)
Tests 726 passed (726)
```

`packages/app` declares Vitest as its package test runner, but this file imports `describe`, `expect`, and `test` from `bun:test`. Vite therefore rejects the Bun-only builtin during collection and the workflow contract executes zero tests.

## Done when

- the contract uses the package's declared Vitest API;
- the focused file is collected and its real workflow assertions pass;
- a mutation or control proves the wrong `bun:test` import reproduces the zero-test failure;
- the full `packages/app` unit suite passes (independent `packages/ui` failures remain owned by #19317);
- package typecheck/Biome and root verification pass on the exact rebased head.

No rendered UI behavior changes, so screenshot/video/OCR evidence is inapplicable.

AI provider/model: openai / gpt-5.6-sol
Client / agent tooling: Codex desktop
Contribution skill revision: N/A - issue triage used repository instructions directly
Attribution status: self-reported
— [codex-issue-triage]
<!-- eliza-computer-attribution:v1 {"provider":"openai","model":"gpt-5.6-sol","client":"Codex desktop","skill_revision":"N/A - issue triage used repository instructions directly"} -->

Implementation and focused validation are complete on `fix/20177-pages-deploy-vitest`.

Current proof after rebasing onto `develop@18d386b87898`:

- Original declared app lane: `packages/app/test/pages-deploy-workflow.test.ts` failed before collection because Vitest cannot import `bun:test`.
- Import-only mutation: still failed before collection because Bun-only `import.meta.dir` is undefined under Vitest.
- Final contract: imports from `vitest` and resolves from `import.meta.dirname`; focused result is 1 file / 2 tests passed.
- Full app scripts: 794 passed, 1 intentional skip, 0 failed.
- Full app Vitest: 89/89 files and 728/728 tests passed.
- App typecheck, changed-file Biome, and `git diff --check` pass.

The exact-head uncached root gate then reached 346/349 tasks before failing only in the independently merged `#20182` cadence widening:

```text
@elizaos/plugin-personal-assistant:typecheck: src/actions/life.ts(4693,15): error TS2345: LifeOpsCadence | undefined is not assignable to LifeOpsCadence.
@elizaos/plugin-personal-assistant:typecheck: src/actions/life.ts(4707,153): error TS2345: LifeOpsCadence | undefined is not assignable to LifeOpsCadence.
@elizaos/plugin-personal-assistant:typecheck: src/actions/life.ts(4789,9): error TS2322: LifeOpsCadence | undefined is not assignable to LifeOpsCadence.
Tasks: 346 successful, 349 total
Cached: 0 cached, 349 total
```

That base regression is tracked and separately owned as #20187. I am not duplicating it. Once its trunk repair lands, I will rebase this one-file patch, rerun the exact-head app/root gates, and publish the PR with the final SHA-bound evidence.

AI provider/model: OpenAI / gpt-5.6-sol
Client / agent tooling: Codex Desktop
Contribution skill revision: N/A - contribute-to-eliza skill was not available in this session
Attribution status: self-reported
— [codex/pages-vitest-contract]
<!-- eliza-computer-attribution:v1 {"provider":"openai","model":"gpt-5.6-sol","client":"Codex Desktop","skill_revision":"N/A - contribute-to-eliza skill was not available in this session"} -->

--- a/packages/app/test/pages-deploy-workflow.test.ts
+++ b/packages/app/test/pages-deploy-workflow.test.ts
@@ -3,12 +3,12 @@
  * Functions worker, exercised without production credentials.
  */
 
-import { describe, expect, test } from "bun:test";
 import { readFileSync } from "node:fs";
 import { join } from "node:path";
+import { describe, expect, test } from "vitest";
 
 const workflow = readFileSync(
-  join(import.meta.dir, "../../../.github/workflows/cloud-cf-release.yml"),
+  join(import.meta.dirname, "../../../.github/workflows/cloud-cf-release.yml"),
   "utf8",
 );
 

